api/src/test/java/com/epam/pipeline/dao/metadata/MetadataEntityDaoTest.java

```diff
@@ -358,35 +358,54 @@ public void testSearch() {
         //filter by field
         MetadataFilter filterByField = createFilter(folder1.getId(), metadataClass1.getName(),
                 Collections.emptyList(),
-                Collections.singletonList(new MetadataFilter.FilterQuery("externalId", EXTERNAL_ID_2)),
+                Collections.singletonList(
+                        new MetadataFilter.FilterQuery("externalId", Collections.singletonList(EXTERNAL_ID_2))),
                 Collections.emptyList(), false);
         checkFilterRequest(filterByField, Collections.singletonList(folder1Sample2));
 
         //filter by json
         MetadataFilter filterByValue = createFilter(folder1.getId(), metadataClass1.getName(),
                 Collections.emptyList(),
-                Collections.singletonList(new MetadataFilter.FilterQuery(DATA_KEY_1, DATA_VALUE_2)),
+                Collections.singletonList(
+                        new MetadataFilter.FilterQuery(DATA_KEY_1, Collections.singletonList(DATA_VALUE_2))),
                 Collections.emptyList(), false);
         checkFilterRequest(filterByValue, Collections.singletonList(folder1Sample2));
 
         MetadataFilter filterByValueRec = createFilter(folder1.getId(), metadataClass1.getName(),
                 Collections.emptyList(),
-                Collections.singletonList(new MetadataFilter.FilterQuery(DATA_KEY_1, DATA_VALUE_1)),
+                Collections.singletonList(
+                        new MetadataFilter.FilterQuery(DATA_KEY_1, Collections.singletonList(DATA_VALUE_1))),
                 Collections.emptyList(), true);
         checkFilterRequest(filterByValueRec, Collections.singletonList(folder1Sample1));
 
         MetadataFilter filterByValueWrongValue = createFilter(folder1.getId(), metadataClass1.getName(),
                 Collections.emptyList(),
-                Collections.singletonList(new MetadataFilter.FilterQuery(DATA_KEY_1, TEST_USER)),
+                Collections.singletonList(
+                        new MetadataFilter.FilterQuery(DATA_KEY_1, Collections.singletonList(TEST_USER))),
                 Collections.emptyList(), true);
         checkFilterRequest(filterByValueWrongValue, Collections.emptyList());
 
         MetadataFilter filterByValueWrongKey = createFilter(folder1.getId(), metadataClass1.getName(),
                 Collections.emptyList(),
-                Collections.singletonList(new MetadataFilter.FilterQuery(DATA_VALUE_2, DATA_VALUE_2)),
+                Collections.singletonList(
+                        new MetadataFilter.FilterQuery(DATA_VALUE_2, Collections.singletonList(DATA_VALUE_2))),
                 Collections.emptyList(), true);
         checkFilterRequest(filterByValueWrongKey, Collections.emptyList());
 
+        MetadataFilter filterByValueSubstring = createFilter(folder1.getId(), metadataClass1.getName(),
```

Comment:
Let's extract a separate tests: one for multiple filter values, one for substring match